### PR TITLE
Add docs for the locking feature

### PR DESCRIPTION
**Pull request by andrejtokarcik**

### CODE REVIEW

File: docs/pages/access-controls/guides/locking.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+<Admonition type="warning">
+  Locking support was introduced in Teleport v7.1. To fully enforce the locks
+  at all components, you should update **all** `teleport` binaries in your
+  deployment.
+</Admonition>
+
+## Creating a new lock
+
+Locks are modelled as [resources](../../admin-guide.mdx#resources) with `kind:
```
Comment (zmb3) [inline docs/pages/access-controls/guides/locking.mdx:30]:
```suggestion
Locks are modeled as [resources](../../admin-guide.mdx#resources) with `kind:
```

File: docs/pages/access-controls/guides/locking.mdx [hunk 2]
```diff
@@ -0,0 +1,106 @@
+---
+title: Locking
+description: Temporarily lock out specific access pathways.
+---
+
+# Session & Identity Locking
```
Comment (klizhentas) [inline docs/pages/access-controls/guides/locking.mdx:6]: Same here `Session and Identity Locking`

File: docs/pages/access-controls/guides/locking.mdx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+Errors returned and warnings logged in this situation feature a message of the
+form:
+```
+lock targeting User:"foo@example.com" is in force: Suspicious activity.
+```
+
+## Locking mode
+
+If a Teleport component cannot properly synchronize its local lock view with
```
Comment (klizhentas) [inline docs/pages/access-controls/guides/locking.mdx:73]: If a teleport node or proxy

File: docs/pages/access-controls/guides/locking.mdx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+the backend, there is a decision to be made about whether to rely on the last
+known locks. This decision strategy is encoded as one of the two modes:
+- `strict` mode causes all interactions to be terminated when the locks are not
+  guaranteed to be up to date;
+- `best_effort` mode keeps relying on the most recent locks.
+
+The cluster-wide mode defaults to `best_effort` and can be set in
+`teleport.yaml` of the Auth server:
+
+```yaml
```
Comment (klizhentas) [inline docs/pages/access-controls/guides/locking.mdx:83]: Can this be done via auth_preference resource as well?

File: docs/pages/access-controls/introduction.mdx [hunk 5]
```diff
@@ -40,6 +40,9 @@ guide.
   <Tile icon="lock" title="Per-session MFA" href="./guides/per-session-mfa.mdx">
     Per-session Multi-Factor Authentication.
   </Tile>
+  <Tile icon="lock" title="Locking" href="./guides/locking.mdx">
+    Session & Identity Locking
```
Comment (klizhentas) [inline docs/pages/access-controls/introduction.mdx:44]: Locking sessions and identities.

File: docs/pages/admin-guide.mdx [hunk 6]
```diff
@@ -959,6 +959,7 @@ Here's the list of resources currently exposed via [`tctl`](cli-docs.mdx#tctl) :
 | cluster | A trusted cluster. See [here](#trusted-clusters) for more details on connecting clusters together. |
 | role | A role assumed by users. The Teleport Open Source Edition only includes one role: "admin", but Enterprise teleport users can define their own roles. |
 | connector | Authentication connectors for [Single Sign-On](enterprise/sso.mdx) (SSO) for SAML, OIDC and Github. |
+| lock | A [lock](./access-controls/guides/locking.mdx) restricting access to the Teleport environment. |
```
Comment (klizhentas) [inline docs/pages/admin-guide.mdx:962]: Locking sessions and identities.

File: docs/pages/access-controls/guides/locking.mdx (hunk 4 above)
Comment (andrejtokarcik) [reply docs/pages/access-controls/guides/locking.mdx:83]: Yes, though that configuration method doesn't seem to be documented anywhere so I followed the convention of the rest of the documentation to show just the `teleport.yaml` snippet.

**Comment by andrejtokarcik** [comment]
@klizhentas I think I've included all your suggestions, please take a look.

**Review by klizhentas — APPROVED** [review]

**Pull request merged**